extensions/amp-video-iframe/1.0/amp-video-iframe.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {BUBBLE_MESSAGE_EVENTS} from '../0.1/amp-video-iframe';
+import {MIN_VISIBILITY_RATIO_FOR_AUTOPLAY} from '../../../src/video-interface';
+import {VideoBaseElement} from '../../amp-video/1.0/base-element';
+import {VideoIframe} from '../../amp-video/1.0/video-iframe';
+import {createCustomEvent} from '../../../src/event-helper';
+import {dict} from '../../../src/utils/object';
+
+/** @const {string} */
+const TAG = 'amp-video-iframe';
+
+/**
+ * @param {!Element} element
+ * @return {!Promise<IntersectionObserverEntry>}
+ */
+function getIntersectionObserverEntryAsync(element) {
+  return new Promise((resolve) => {
+    const observer = new IntersectionObserver((entries) => {
+      resolve(entries[entries.length - 1]);
+      observer.disconnect();
+    });
+    observer.observe(element);
+  });
+}
+
+/**
+ * @param {!Element} element
+ * @return {!Promise<number>}
+ */
+function getIntersectionRatioMinAutoplay(element) {
+  return getIntersectionObserverEntryAsync(element).then(
+    ({intersectionRatio}) =>
+      // Only post ratio > 0 when in autoplay range to prevent internal
+      // autoplay implementations that differ from ours.
```

Comment:
Not sure if this detail is important anymore, since the new Classic implementation using `InObs` doesn't care about this ratio (this should probably change in the Preact implementation too).

Reply:
To clarify: this is the minimum intersection ratio at which we report the intersection ratio? This doesn't quite make sense to me.

Reply:
Yeah, we limit the posted intersection to [50%, 100%], where anything below 50% is 0. The original thinking for this might not be relevant anymore.